python/ray/tests/test_autoscaler.py

```diff
@@ -3514,14 +3514,18 @@ class FaultyAutoscaler:
             def __init__(self, *args, **kwargs):
                 raise AutoscalerInitFailException
 
+        os.environ["RAY_GCS_SERVER_PORT"] = "12345"
+        ray.init()
```

Comment:
consider using fixture here. ray init won't be clean up after this test.

(or at least with manager)

Reply:
Apparently pytest fixtures don't work with the unittest package, and the usual way to do this seems to be in the setup and teardown methods. Since this is specific to this particular method, I just unset it afterwards. We can clean up the test in a followup.